### PR TITLE
feat: Tip Claim frontend integration

### PR DESCRIPTION
**Pull request by ugur-eren**
This PR integrates the new backend endpoint to the frontend for tip claiming. It also adds a new endpoint for fee calculation.

Endpoints:
- POST estimate-claim - body: {event: NostrEvent} - response: {fee: string}
- POST claim - body: {event: NostrEvent} - response: {transaction_hash: string}

### CODE REVIEW

File: JoyboyCommunity/src/screens/Tips/index.tsx [hunk 1]
```diff
@@ -29,77 +30,55 @@ export const Tips: React.FC = () => {
...
+        connectedAccount.address,
+      )},${cairo.felt(STRK[chainId].address)},${gasAmount.toString()}`;
+      event.tags = [];
+
+      await event.sign();
+      return event.rawEvent();
+    };
+
+    const feeResult = await estimateClaim.mutateAsync(
+      await getNostrEvent(BigInt(0.0004 * Number(decimalsScale(18)))),
```
Comment (maciejka) [inline JoyboyCommunity/src/screens/Tips/index.tsx:61]: Where this constant come from? Does it take into account gas token price fluctuations?

File: website/src/app/api/deposit/estimate-claim/route.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  } catch (error) {
+    if (error instanceof Error) {
+      return NextResponse.json({code: error.message}, {status: HTTPStatus.BadRequest});
+    }
+
+    throw error;
+  }
+
+  try {
+    const result = await account.estimateInvokeFee([
```
Comment (maciejka) [inline website/src/app/api/deposit/estimate-claim/route.ts:38]: In the end does it need a signature or not?

Comment (ugur-eren) [reply website/src/app/api/deposit/estimate-claim/route.ts:38]: For infura it does, for others it doesn't. However starknet.js signs the transaction as a real invocation, not as an estimation, so opting-out and not using signature.

File: JoyboyCommunity/src/screens/Tips/index.tsx (hunk 1 above)
Comment (ugur-eren) [reply JoyboyCommunity/src/screens/Tips/index.tsx:61]: It is only used during fee estimation, I didn't want it to be 0 during the estimation to correctly estimate the fees.

**Pull request merged**